src/transformers/models/opt/modeling_opt.py

```diff
@@ -642,11 +862,15 @@ def forward(
         # embed positions
         if attention_mask is None:
             attention_mask = torch.ones(batch_size, mask_seq_length, device=inputs_embeds.device)
+            padding_mask = None
         elif attention_mask.shape[1] != mask_seq_length:
             raise ValueError(
                 f"The provided attention mask has length {attention_mask.shape[1]}, but its length should be "
                 f"{mask_seq_length} (sum of the lengths of current and past inputs)"
             )
+        else:
+            padding_mask = attention_mask
```

Comment:
Can you try to add a conditional check to check if there is any 0 in the attention mask, sometimes users pass an attention mask with a full ones, therefore we'll need to set `padding_mask` to None to avoid entering the padding case in `OPTFlashAttention2` module

If the test fails with torch.fx can you try with `torch.isin` ? otherwise you can also do a check based on the sum of the attention mask

Reply:
Ok, sorry for removing it directly without discussing.

Reply:
no problem at all @susnato ! 🙏

Reply:
It fails for - 
- `torch.isin(attention_mask.int(), 0).sum().item()!=0`(AssertionError: Couldn't trace module: symbolically traced variables cannot be used as inputs to control flow
) 
- `torch.isin(attention_mask.int(), 0).sum()!=0`(AssertionError: Couldn't trace module: __bool__ should return bool, returned Tensor
)
- `attention_mask.sum().item()==attention_mask.numel()`(AssertionError: Couldn't trace module: symbolically traced variables cannot be used as inputs to control flow
)
- `if not torch.equal(attention_mask, torch.ones_like(attention_mask))`( AssertionError: Couldn't trace module: symbolically traced variables cannot be used as inputs to control flow)

Reply:
There seems to be a related open issue [here](https://github.com/pytorch/pytorch/issues/54602)